src/machine.rs

```diff
@@ -66,17 +72,17 @@ pub struct Frame<'tcx> {
 impl<'tcx> Frame<'tcx> {
     pub fn new(def_id: DefId,
                mir: &'tcx Mir<'tcx>,
-               ret_addr: Option<BasicBlock>)
+               locals: IndexVec<Local,Option<Vec<u8>>>,
+               ret_val: Option<Address>,
+               ret_block: Option<BasicBlock>)
                -> Frame<'tcx> {
-        let size = mir.local_decls.len();
-        let mut locals:IndexVec<Local, Option<Vec<u8>>> = IndexVec::with_capacity(size);
-        locals.extend(vec![None; size]);
 
         Frame {
```

Comment:
For future reference, you can use `Self { ... }` which makes the code easier to maintain (e.g. easier to rename structs later).

Reply:
That's neat. I didn't know this!